manifests/init.pp

```diff
@@ -1,49 +1,91 @@
-#   Specify version of nomad binary to download.
-#
-# [*config_hash*]
-#   Use this to populate the JSON config file for nomad.
-#
-# [*config_mode*]
-#   Use this to set the JSON config file mode for nomad.
-#
-# [*pretty_config*]
-#   Generates a human readable JSON config file. Defaults to `false`.
-#
-# [*pretty_config_indent*]
-#   Toggle indentation for human readable JSON file. Defaults to `4`.
-#
-# [*install_method*]
-#   Valid strings: `package` - install via system package
-#                  `url`     - download and extract from a url. Defaults to `url`.
-#                  `none`    - disable install.
-#
-# [*package_name*]
-#   Only valid when the install_method == package. Defaults to `nomad`.
-#
-# [*package_ensure*]
-#   Only valid when the install_method == package. Defaults to `installed`.
-#
-#
-# [*restart_on_change*]
-#   Determines whether to restart nomad agent on $config_hash changes.
-#   This will not affect reloads when service, check or watch configs change.
-# Defaults to `true`.
+# @example To set up a single nomad server, with several agents attached, on the server.
+#   class { '::nomad':
+#     version     => '1.0.1', # check latest version at https://github.com/hashicorp/nomad/blob/master/CHANGELOG.md
+#     config_hash => {
+#       'region'     => 'us-west',
+#       'datacenter' => 'ptk',
+#       'log_level'  => 'INFO',
+#       'bind_addr'  => '0.0.0.0',
+#       'data_dir'   => '/opt/nomad',
+#       'server'     => {
+#         'enabled'          => true,
+#         'bootstrap_expect' => 3,
+#       }
+#     }
+#   }
 #
-# [*extra_options*]
-#   Extra arguments to be passed to the nomad agent
+# @example On the agent(s)
+#   class { 'nomad':
+#     config_hash => {
+#         'region'     => 'us-west',
+#         'datacenter' => 'ptk',
+#         'log_level'  => 'INFO',
+#         'bind_addr'  => '0.0.0.0',
+#         'data_dir'   => '/opt/nomad',
+#         'client'     => {
+#         'enabled'    => true,
+#         'servers'    => [
+#           "nomad01.your-org.pvt:4647",
+#           "nomad02.your-org.pvt:4647",
+#           "nomad03.your-org.pvt:4647"
+#         ]
+#       }
+#     },
+#  }
 #
-# [*init_style*]
-#   What style of init system your system uses.
+# @example Disable install and service components
+#   class { '::nomad':
+#     install_method => 'none',
+#     init_style     => false,
+#     manage_service => false,
+#     config_hash   => {
+#       'region'     => 'us-west',
+#       'datacenter' => 'ptk',
+#       'log_level'  => 'INFO',
+#       'bind_addr'  => '0.0.0.0',
+#       'data_dir'   => '/opt/nomad',
+#       'client'     => {
+#       'enabled'    => true,
+#       'servers'    => [
+#           "nomad01.your-org.pvt:4647",
+#           "nomad02.your-org.pvt:4647",
+#           "nomad03.your-org.pvt:4647"
+#         ]
+#       }
+#     },
+#  }
 #
-# [*purge_config_dir*]
-#   Purge config files no longer generated by Puppet
+# @param arch cpu architecture
```

Comment:
I like to split params on 2 lines. It especially helps on longer lines
```suggestion
# @param arch
#    cpu architecture
```

Reply:
ok i ll reformat